Add feature for fetching all JSON secret properties.

iAnomaly opened the pull request.

arruzk commented:
@iAnomaly this change has a bug when you specify `property` option.
```
aws secretsmanager create-secret --name hello-service/credentials --secret-string '{"username":"admin","password":"1234"}'
```
```
apiVersion: 'kubernetes-client.io/v1'
kind: ExternalSecret
metadata:
  name: test-secrets-notifier
secretDescriptor:
  backendType: secretsManager
  data:
    - key: hello-service/credentials
      name: password
      property: password
```
Result is (instead of `1234`):
```
{
    "apiVersion": "v1",
    "data": {
        "0": "MQ==",
        "1": "Mg==",
        "2": "Mw==",
        "3": "NA=="
    },
    "kind": "Secret",
...
```

iAnomaly commented:
@arruzk Thanks for finding and reporting but I cannot reproduce no matter how hard I try. What version of node did you test this on? Only thing I can think of is Objects.keys changed for newer versions of ECMA so I switched to `Object.getOwnPropertyNames` if you don't mind testing again.

Also expanded the test coverage and fixed one bug corner case (not the one you reported).

arruzk commented:
@iAnomaly thank you! 

Now it's working fine if I specify both `name` and `property`. 
I have an error if I specify only `property` (without `name`), but probably it should be like this (maybe error can be a little more verbose)
```
TypeError [ERR_INVALID_ARG_TYPE]: The first argument must be one of type string, Buffer, ArrayBuffer, Array, or Array-like Object. Received type object\n    at Function.from (buffer.js:233:9)\n    at Object.getOwnPropertyNames.forEach.key (/Users/u/projects/tmp/kubernetes-external-secrets/lib/backends/kv-backend.js:82:29)\n    at Array.forEach (<anonymous>)\n    at externalData.forEach (/Users/u/projects/tmp/kubernetes-external-secrets/lib/backends/kv-backend.js:81:63)\n    at Array.forEach (<anonymous>)\n    at SecretsManagerBackend.getSecretManifestData (/Users/u/projects/tmp/kubernetes-external-secrets/lib/backends/kv-backend.js:75:18)\n    at process._tickCallback (internal/process/next_tick.js:68:7)","msg":"failure while polling the secret test-secrets-notifier","v":1}
```
```
  data:
    - key: staging/secrets-common
      property: SECRET_HASH_SALT
```
`secretPropertyValues` object is 
```
[{ undefined: { SECRET_HASH_SALT: 'salt' } }]
```
Node version: v10.13.0
Npm version: 6.4.1

iAnomaly commented:
Thank you @arruzk, I could reproduce this one. It is now fixed with added test coverage for this case. Appreciate your testing!

Flydiverny commented:
To add this I think we should do something similar to what was discussed here https://github.com/godaddy/kubernetes-external-secrets/issues/94#issuecomment-505290048

Flydiverny commented:
I took the liberty of taking a pass on this in #195 ( based on #94 )

iAnomaly commented:
Closed in favor of #196.
